highfidelity/hifi-content
# Adding mirror to marketplace

Finally adding the mirror to the marketplace! Woot!

# Test Plan
## Prerequisites
You should only need the latest dev download of [High Fidelity](https://highfidelity.com/dev-download)
## Tests

1. Turn on Advanced Menus by selecting "Settings -> Advanced Menus" on the menu bar.
2. Click "Edit -> Import Entities from URL" on the menu bar.
3. Copy and paste this link into the import field: https://raw.githubusercontent.com/patrickmanalich/hifi-content/hifi-content_mirror/Marketplace/mirror/m.json
4. Click "okay" and check to see if a mirror imports in front of you.
5. Check to see if a button has appeared near the top right corner of the mirror.
6. Check to see if a blue box has appeared at the lower right corner of the mirror.
7. Click the button with your mouse and ensure you see your avatar being rendered in the mirror, and that the button has changed to a red color.
8. Drag the blue box in multiple directions. See if the mirror stretches with the box, and the rendering stays aligned with the mirror. Also check that the box has a highlighted tint to it.
9. Let go of the blue box and make sure that box loses its highlighted tint and that the avatar is still being rendered.
10. Stretch the box as far and as small as you can, and make sure that the mirror has limits to how large and how small you can resize it.
11. Click the button with your mouse and ensure that you no longer see your avatar being rendered in the  mirror, and that the button has changed back to its grey color.
12. Go into HMD.
13. Laser trigger the button, and make sure that the mirror begins rendering (same as the mouse click).
14. Grab the blue box with you hand controller, and ensure that the dragging mechanism still works.
15. Laser trigger the button again, making sure the rendering stops.
16. Import the Spectator Camera from Marketplace and turn it on.
17. Attempt to turn on the mirror and make sure the mirror doesn't start rendering (it shouldn't turn on).
16. Delete the mirror and make sure both the button and the blue box are deleted with it.
